### PR TITLE
fix(select) select drops outside the window, closes #284

### LINKED ISSUE #284: select drops outside the window
**I'm submitting a bug report**

* **Library Version:**
0.20.0

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Node Version:**
12.16.2

* **NPM Version:**
6.14.4

* **JSPM OR Webpack AND Version**
webpack 4.42.1

* **Browser:**
all

* **Language:**
TypeScript 3.9

**Current behavior:**
When select is close to the window botton the dropdown drops outside the window

**Expected/desired behavior:**
* **What is the expected behavior?**
Either scroll is enabled or dropdown moves above the element

* **What is the motivation / use case for changing the behavior?**
Support different select positions on the page

### PR DESCRIPTION
**Pull request by matthewcorven**

### CODE REVIEW

File: packages/select/src/ux-select.ts [hunk 1]
```diff
@@ -261,15 +264,19 @@ export class UxSelect implements UxInputComponent {
...
     this.winEvents.disposeAll();
   }
 
-  public listAnchor: { x: number | string, y: number | string } | null;
-  private calcAnchorPosition() {
-    const elDim = this.element.getBoundingClientRect();
-    const offsetY = (48 - elDim.height) / 2;
-    this.listAnchor = { x: elDim.left, y: elDim.top - offsetY };
+  private setListAnchorPosition() {
+    this.positioningFactory(
```
Comment (bigopon) [inline packages/select/src/ux-select.ts:271]: calling the factory function will instantiate a new instance of `Positioning` every time. Is this intentional?

File: packages/select/src/ux-select.ts [hunk 2]
```diff
@@ -261,15 +264,19 @@ export class UxSelect implements UxInputComponent {
...
-    this.listAnchor = null;
     this.winEvents.disposeAll();
   }
 
-  public listAnchor: { x: number | string, y: number | string } | null;
-  private calcAnchorPosition() {
-    const elDim = this.element.getBoundingClientRect();
-    const offsetY = (48 - elDim.height) / 2;
-    this.listAnchor = { x: elDim.left, y: elDim.top - offsetY };
+  private setListAnchorPosition() {
```
Comment (bigopon) [inline packages/select/src/ux-select.ts:270]: The "set" position job is actually done by the positioning service, so maybe we need a method name to reflect that, or at least avoid giving wrong impression? I think we can just stop at `updateListAnchorPosition`

Comment (matthewcorven) [reply packages/select/src/ux-select.ts:270]: Good call, I'll change it

File: packages/select/src/ux-select.ts (hunk 1 above)
Comment (matthewcorven) [reply packages/select/src/ux-select.ts]: I borrowed the approach from model demo
https://github.com/aurelia/ux/blob/b19034779fa081ca5ce744eafa4cb44c0dc723b2/app/src/modal-page.ts#L110

Moved to `attached`

**Comment by bigopon** [comment]
thanks @matthewcorven 👍

**Pull request merged**